File: Java_Collections_Benchmark/src/com/company/HashSetTest.java
package com.company;

import java.util.HashSet;

public class HashSetTest implements IBenchmark {

    HashSet<Integer> hashSet;
    int numOfOperations;

    public HashSetTest(){
        hashSet = new HashSet<>();
        this.numOfOperations = 100;
    }

    public HashSetTest(int numOfOperations){
        hashSet = new HashSet<>();
        this.numOfOperations = numOfOperations;
    }

    @Override
    public void AddTest() {
        for (int i = 0; i < numOfOperations; i++) {
            hashSet.add(i);
        }
    }

    @Override
    public void FindTest() {
        Integer toFind = hashSet.size() - 1;
        for (int i = 0; i < numOfOperations; i++) {
            boolean result = hashSet.contains(toFind);
        }
    }

    @Override
    public void RemoveTest() {
        Integer toRemove = hashSet.size() - 1;
        for (int i = 0; i < numOfOperations; i++) {
            hashSet.remove(toRemove);
            toRemove--;
        }
    }
}

File: CS_Collections_benchmark/HashSetTest.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CS_Collections_benchmark
{
    class HashSetTest : IBenchmark
    {
        HashSet<int> hashSet;
        int numOfOperations;
        public HashSetTest()
        {
            hashSet = new HashSet<int>();
            numOfOperations = 100;
        }

        public HashSetTest(int numOfOperations)
        {
            hashSet = new HashSet<int>();
            this.numOfOperations = numOfOperations;
        }


        public void AddTest()
        {
            for (int i = 0; i < numOfOperations; i++)
            {
                hashSet.Add(i);
            }
        }

        public void FindTest()
        {
            int toFind = hashSet.Count - 1;
            for (int i = 0; i < numOfOperations; i++)
            {
                bool result = hashSet.Contains(toFind);
            }
        }

        public void RemoveTest()
        {
            int toRemove = hashSet.Count - 1;
            for (int i = 0; i < numOfOperations; i++)
            {
                hashSet.Remove(toRemove);
                toRemove--;
            }

        }
    }
}
